### DISCUSSION TITLE
v4.1.5 pre-release

### DISCUSSION THREAD

**Discussion by autoSteve** [BJReplay/ha-solcast-solar#157] [Announcements]
## What's Changed

A bug fixed based on feedback from the previous pre-release (thank you!), plus a couple more we found.

* Bug: Missing iterator in new sites check by @autoSteve
* Bug: Timestamp stored in usage cache was wrong by @autoSteve
* Bug: Adding API key reset usage for first key by @autoSteve
* Work around a possible HA scheduling bug by @autoSteve
* Code style alignment to HA style guidelines by @autoSteve

Full Changelog: https://github.com/BJReplay/ha-solcast-solar/compare/v4.1.4...v4.1.5

**Comment by autoSteve** [comment]
As has become tradition, notifying the usual early-adopting-and-reporting folk. @dtswk, @EddieSpaghettie, @ffm777, @gcoan, @hwb45, @iainfogg, @isorin, @jeffb1974, @kernow, @ladaan, @Markyt77, @Nilogax, @Noodleyman, @ProphetOfDoom, @rholligan, @sh00t2kill, @Unlik and @wimdebruyn.

Happy hump day.

**Comment by gcoan** [comment]
Installed 4.1.5, all seems to be working great.

Two observations:

1. In INFO logging mode can an extra line be added to the logging to indicate when the sensor update has completed successfully as currently you can see that the logging starts but not ends:

```

2024-09-06 01:01:00.313 INFO (MainThread) [custom_components.solcast_solar] Service call: update_forecasts
2024-09-06 01:01:00.314 INFO (MainThread) [custom_components.solcast_solar.solcastapi] Getting forecast update for Solcast site bc31-4d4c-e844-6a4e
2024-09-06 01:01:00.380 WARNING (MainThread) [custom_components.solcast_solar.solcastapi] The Solcast API is busy, pausing 19 seconds before retry
2024-09-06 01:01:19.428 WARNING (MainThread) [custom_components.solcast_solar.solcastapi] The Solcast API is busy, pausing 33 seconds before retry
2024-09-06 01:01:52.488 WARNING (MainThread) [custom_components.solcast_solar.solcastapi] The Solcast API is busy, pausing 50 seconds before retry
2024-09-06 01:02:43.812 INFO (MainThread) [custom_components.solcast_solar.solcastapi] Getting forecast update for Solcast site 23bb-0216-85c6-fd94
```

Probably just needs a Debug message reclassifying as Info type

2. state_class comment made on 4.1.4 (but seen first on 4.1.3) https://github.com/BJReplay/ha-solcast-solar/discussions/150#discussioncomment-10513137

**Reply by autoSteve** [reply]
> 1.

Like thusly? Good idea.

```
INFO (MainThread) [custom_components.solcast_solar] Service call: update_forecasts
INFO (MainThread) [custom_components.solcast_solar.solcastapi] Getting forecast update for Solcast site b68d-c05a-c2b3-2cf9
INFO (MainThread) [custom_components.solcast_solar.solcastapi] Getting forecast update for Solcast site 83d5-ab72-2a9a-2397
INFO (MainThread) [custom_components.solcast_solar.solcastapi] Forecast update completed successfully
```

> 2.

What would be the impact of the change? LTS for these sensors would probably go AWOL, which I don't care about, but do others?

**Reply by gcoan** [reply]
1. thanks

2. dunno what it would do to LTS.  Only one way to find out, set it up on a test rig and see how Home Assistant handles it.

There may be different results for those sensors that don't currently have a class and for those that I propose the class is wrong.

**Comment by autoSteve** [comment]
v4.1.6 "latest" is published, so closing this discussion.

> Only one way to find out, set it up on a test rig and see how Home Assistant handles it.

On the "to do" list, @gcoan.
